### PR TITLE
osdc: add timeout configs for mons/osds

### PR DESCRIPTION
**Pull request by batrick**
DNM because I want to test this with Octopus before merging. The problem is avoided on testing in master.

See: https://tracker.ceph.com/issues/47734

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
retest this please

**Comment by batrick** [comment]
https://github.com/ceph/ceph/pull/37530#issuecomment-704514437

**Comment by batrick** [comment]
jenkins test api

**Review by dillaman — DISMISSED** [review]
osdc changes lgtm (PR has been intermixed with apparent unrelated changes per the PR title, though)

**Comment by batrick** [comment]
retest this please

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201007.214100

failures unrelated.

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
@ceph/dashboard please take a look at this API test failure I can't seem to get rid of. I don't think it's caused by this PR.

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test make check

File: src/osdc/Objecter.h [hunk 1]
```diff
@@ -2446,7 +2446,9 @@ class Objecter : public md_config_obs_t, public Dispatcher {
 				 boost::system::error_code>>> waiting_for_map;
 
   ceph::timespan mon_timeout;
+  bool mon_timeout_configured = false;
```
Comment (dillaman) [inline src/osdc/Objecter.h:2449]: Code keeps changing since my last review -- but these are no longer used in your latest version.

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
@dillaman this is ready for a final review. I've sorted out the API test failures and cleaned up the commits.

**Review by dillaman — APPROVED** [review]
lgtm

**Comment by s0nea** [comment]
The dashboard QA test run was successful: https://pulpito.ceph.com/tdehler-2020-10-16_09:25:17-rados:dashboard-wip-tdehler-testing-37529-37564-distro-basic-smithi/

**Review by jdurgin — APPROVED** [review]

**Review by neha-ojha — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201013.174240

failures unrelated

**Comment by neha-ojha** [comment]
@batrick https://tracker.ceph.com/issues/48030 is a regression introduced by this PR, @sseshasa has root-caused it here https://tracker.ceph.com/issues/48030#note-11.

**Comment by batrick** [comment]
> @batrick https://tracker.ceph.com/issues/48030 is a regression introduced by this PR, @sseshasa has root-caused it here https://tracker.ceph.com/issues/48030#note-11.

This issue is really caused by this commit 306eebe05bec1220b2477a66b6501da222bacdfc. The RadosClient seemed to be plugged into the conf change interface as an observer but the `add_observer` call in the cons was forgotten.

Of course, the changes in this PR exposed the problem so I'll fix it.

**Pull request merged**